COMETwebapp/COMETwebapp/SessionManagement/SessionAnchor.cs

```diff
@@ -35,7 +37,52 @@ public class SessionAnchor : ISessionAnchor
         /// <summary>
         /// Gets or sets the <see cref="ISession"/>
         /// </summary>
-        public ISession? Session { get; set; }
+        public ISession Session { get; set; }
 
+        /// <summary>
+        /// True if the <see cref="ISession"/> is opened
+        /// </summary>
+        public bool IsSessionOpen { get; set; }
+
+        /// <summary>
+        /// The opened <see cref="Iteration"/>
+        /// </summary>
+        public Iteration OpenIteration { get; set; }
+
+        /// <summary>
+        /// The <see cref="DomainOfExpertise"/> selected to open a model
+        /// </summary>
+        public DomainOfExpertise CurrentDomainOfExpertise { get; set; }
+
+        /// <summary>
+        /// Retrieves the <see cref="SiteDirectory"/> that is loaded in the <see cref="ISession"/>
+        /// </summary>
+        /// <returns>The <see cref="SiteDirectory"/></returns>
+        public SiteDirectory GetSiteDirectory() => this.Session.RetrieveSiteDirectory();
+
+        /// <summary>
+        /// Open the iteration with the selected <see cref="EngineeringModelSetup"/> and <see cref="IterationSetup"/>
+        /// </summary>
+        /// <param name="modelSetup"> The selected <see cref="EngineeringModelSetup"/> </param>
+        /// <param name="iterationSetup">The selected <see cref="IterationSetup"/></param>
+        public async void GetIteration(EngineeringModelSetup modelSetup, IterationSetup iterationSetup)
+        {
+            var model = new EngineeringModel(modelSetup.EngineeringModelIid, this.Session.Assembler.Cache, this.Session.Credentials.Uri);
+            var iteration = new Iteration(iterationSetup.IterationIid, this.Session.Assembler.Cache, this.Session.Credentials.Uri);
+            iteration.Container = model;
+
+            await this.Session.Read(iteration, this.CurrentDomainOfExpertise);
```

Comment:
I think we need to add some `try catch finally` statements here to be able to recover from errors and also show an error message that makes sense to the user. Please have a look at the IME to get inspiration.